corehq/apps/data_interfaces/migrations/0033_automaticupdaterule_deleted_on.py

```diff
@@ -0,0 +1,31 @@
+from django.db import migrations, models
+
+from corehq.apps.cleanup.utils import migrate_to_deleted_on
+from corehq.apps.data_interfaces.models import AutomaticUpdateRule
+
+INDEX_NAME = 'automaticupdaterule_deleted_on_idx'
+TABLE_NAME = 'data_interfaces_automaticupdaterule'
+COLUMNS = 'deleted_on'
+WHERE_CLAUSE = 'deleted_on IS NOT NULL'
+CREATE_INDEX = f"CREATE INDEX {INDEX_NAME} ON {TABLE_NAME} ({COLUMNS}) WHERE {WHERE_CLAUSE}"
+
+
+def set_deleted_on_field(apps, schema_editor):
+    migrate_to_deleted_on(AutomaticUpdateRule, 'deleted', should_audit=True)
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('data_interfaces', '0032_bootstrap_audit_events_for_update_rules'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='automaticupdaterule',
+            name='deleted_on',
+            field=models.DateTimeField(null=True),
+        ),
+        migrations.RunSQL(CREATE_INDEX),
```

Comment:
It's possible to create partial indexes in Django models, which are then automatically detected by the migration framework. Example:

https://github.com/dimagi/commcare-hq/blob/5c59cd06fa808515f3d46ba1282767eb8ee801f2/corehq/blobs/models.py#L77-L81

Reply:
d36834a865b650d30d5bad35a814c18557e1af23